### PR TITLE
create room form

### PR DESCRIPTION
**Pull request by thucpv-1748**
https://edu-redmine.sun-asterisk.vn/issues/9214

### CODE REVIEW

File: app/Http/Controllers/Backend/RoomController.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers\Backend;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Repositories\Contracts\StoreInterface;
```
Comment (framgiaciv3) [inline app/Http/Controllers/Backend/RoomController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse</br>**Severity**: error</br>**File**: app/Http/Controllers/Backend/RoomController.php</br>**Line**: 9</br>There must be one blank line after the last USE statement; 2 found;

File: app/Http/Controllers/Backend/RoomController.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    /**
+     * RoomController constructor.
+     * @param RoomInterface $model
+     * @param StoreInterface $store
+     * @param SeatInterface $seat
+     */
+    public function __construct(
+        RoomInterface $model,
+        StoreInterface $store,
+        SeatInterface $seat
```
Comment (framgiaciv3) [inline app/Http/Controllers/Backend/RoomController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.NewlineBeforeOpenBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/Backend/RoomController.php</br>**Line**: 39</br>The closing parenthesis and the opening brace of a multi-line function declaration must be on the same line

File: app/Http/Controllers/Backend/RoomController.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    {
+        $this->model = $model;
+        $this->store = $store;
+        $this->seat = $seat;
+    }
+
+    /**
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function getRoom()
```
Comment (framgiaciv3) [inline app/Http/Controllers/Backend/RoomController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.Incorrect</br>**Severity**: error</br>**File**: app/Http/Controllers/Backend/RoomController.php</br>**Line**: 50</br>Line indented incorrectly; expected at least 8 spaces, found 7

File: app/Http/Controllers/Backend/RoomController.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    {
+       $room = $this->model->with('store')->paginate(config('setting.paginate'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/Backend/RoomController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.ScopeIndent.Incorrect</br>**Severity**: error</br>**File**: app/Http/Controllers/Backend/RoomController.php</br>**Line**: 52</br>Line indented incorrectly; expected at least 8 spaces, found 7

File: app/Http/Controllers/Backend/RoomController.php [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+        return view('layout/backend/roomform', compact('store', 'room', 'seat'));
+    }
+
+    /**
+     * @param $id
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function deleteRoom($id)
+    {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Backend/RoomController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/Backend/RoomController.php</br>**Line**: 122</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Repositories/RoomRepository.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        $data = $this->update($input->id, $input->all());
+        $this->saveSeatByRoom($data->id, $input->row);
+
+        return $data;
+    }
+
+    /**
+     * @param $roomid
+     * @param $input
```
Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 55</br>Opening brace should be on a new line

File: app/Repositories/RoomRepository.php [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    public function saveSeatByRoom($roomid, $input){
+        $this->seat->where('room_id', $roomid)->delete();
```
Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ForEachLoopDeclaration.SpaceBeforeClose</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 58</br>Space found before closing bracket of FOREACH loop

Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.ControlStructureSpacing.SpaceBeforeCloseBrace</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 58</br>Expected 0 spaces before closing bracket; 1 found

File: app/Repositories/RoomRepository.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        if ($input) {
+            foreach ($input as $key => $value ) {
```
Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 60</br>String "row" does not require double quotes; use single quotes instead

File: app/Repositories/RoomRepository.php [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+                $data = [
```
Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 61</br>String "col" does not require double quotes; use single quotes instead

File: app/Repositories/RoomRepository.php [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+                    "row" => $key,
```
Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 62</br>String "room_id" does not require double quotes; use single quotes instead

File: app/Repositories/RoomRepository.php [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        return $data;
+    }
+
+    /**
+     * @param $roomid
+     * @param $input
+     */
+    public function saveSeatByRoom($roomid, $input)
+    {
+        $this->seat->where('room_id', $roomid)->delete();
```
Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 59</br>Expected 1 space(s) after FOREACH keyword; 0 found

Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ForEachLoopDeclaration.SpaceBeforeClose</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 59</br>Space found before closing bracket of FOREACH loop

Comment (framgiaciv3) [inline app/Repositories/RoomRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.ControlStructureSpacing.SpaceBeforeCloseBrace</br>**Severity**: error</br>**File**: app/Repositories/RoomRepository.php</br>**Line**: 59</br>Expected 0 spaces before closing bracket; 1 found

File: app/Http/Controllers/Backend/RoomController.php [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+
+    /**
+     * @param $id
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function editRoom($id)
+    {
+        $store =  $this->store->all();
```
Comment (BunnyPi04) [inline app/Http/Controllers/Backend/RoomController.php]: `$store =  $this->store->all();`
dư 1 space

File: resources/assets/css/backend/room.scss [hunk 13]
```diff
@@ -0,0 +1,27 @@
+.seat {
+  position: absolute;
```
Comment (BunnyPi04) [inline resources/assets/css/backend/room.scss]: indent = 4 spaces

File: resources/views/layout/backend/roomform.blade.php [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+                    <label for="name">{{ __('Seats in row') }} 1:</label>
+                    <input type="number" class="form-control" name="row[]" value="" required="required">
+                </div>
+            @else
+                @php($x = 1)
+                @foreach($seat as $key => $data)
+                        <div class="form-group">
+                            <label for="name">{{ __('Seats in row') . $x }}:</label>
+                            <input type="number" class="form-control" name="row[]" value="{{ $data['col'] }}" required="required">
+                            @if($x > 1)
```
Comment (BunnyPi04) [inline resources/views/layout/backend/roomform.blade.php]: convention

File: resources/views/layout/backend/roomform.blade.php [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+        </form>
+        <div class="seat">
+            @if (count($seat) > 0)
+                <h2>{{ __('Screen') }}</h2>
+               @foreach ($seat as $value)
+                   <ul>
+                       <p>{{ Config('nameRow.' . $value['row']) }}</p>
+                       @for ($x = 0; $x < $value['col']; $x++)
+                           <li>
+                                {{ $x+1 }}
```
Comment (BunnyPi04) [inline resources/views/layout/backend/roomform.blade.php]: convention

File: resources/assets/css/backend/room.scss [hunk 16]
```diff
@@ -0,0 +1,27 @@
+.seat {
+    position: absolute;
+    left: 50%;
+    transform: translatex(-50%);
+    padding-bottom: 100px;
+    li {
+      margin: 5px;
```
Comment (BunnyPi04) [inline resources/assets/css/backend/room.scss]: indent = 4 spaces

**Pull request merged**